javascript/javascript1/week2/flight-booking.js

```diff
@@ -0,0 +1,41 @@
+// Flight booking fullname function
+
+// When a user books a flight they write their firstname and surname, but when the ticket is printed a fullname should be displayed.
+
+// Created a function which generate a full name from firstName, surname and useFor
+function getFullName(firstName, surname, useFormalName, isMale) {
+  // First statement is checking whether given parameters are strings or not and being sure they are not empty
+  if (
+    typeof firstName && //check parameter type
+    typeof surname === "string" && //checks parameter type
+    surname.trim().length !== 0 && //removing all the spaces to check if its empty string or not
+    firstName.trim().length !== 0 //removing all the spaces to check if its empty string or not
+  ) {
+    // Second statement is checking if useFormalName parameter is true or false for giving the lord title
+    if (useFormalName) {
+      // Third statement is checking whether man or woman
+      if (!isMale) {
+        return `Lord ${firstName} ${surname}`;
```

Comment:
You repeat `${firstName} ${surname}` three times. There is a coding principle called [DRY (Don't Repeat Yourself) ](https://en.wikipedia.org/wiki/Don%27t_repeat_yourself)which suggests creating variables and functions whenever you are to repeat a block of code more than once.

```
const fullName = `${firstName} ${surname}`;
```